src/ics_cal/icshelper.py

```diff
@@ -23,52 +23,57 @@ def retrieve_events(self, ics_url, calStartDatetime, calEndDatetime, localTZ):
         event_list = []
 
         self.logger.info("Retrieving events from ICS...")
-        response = requests.get(ics_url)
-        if response.ok:
-            cal = icalendar.Calendar.from_ical(response.text)
-        else:
-            self.logger.error(f"Received an error when downloading ICS: {response.text}")
-            sys.exit(1)
-
-        events = recurring_ical_events.of(cal).between(calStartDatetime, calEndDatetime)
-        local_timezone = pytz.timezone(localTZ)
-
-        for event in events:
-            new_event = {"summary": str(event.get("SUMMARY"))}
-
-            if "LOCATION" in event:
-                new_event["location"] = str(event.get("LOCATION"))
-
-            event_start = event.get("DTSTART").dt
-            event_end = event.get("DTEND").dt
-
-            if isinstance(event_start, dt.datetime):
-                new_event["allday"] = False
-                new_event["startDatetime"] = event_start.astimezone(local_timezone)
-                new_event["endDatetime"] = event_end.astimezone(local_timezone)
-            elif isinstance(event_start, dt.date):
-                new_event["allday"] = True
-                # Convert date into datetime at midnight
-                new_event["startDatetime"] = local_timezone.localize(
-                    dt.datetime.combine(event_start, dt.time(0, 0))
-                )
-                new_event["endDatetime"] = local_timezone.localize(
-                    dt.datetime.combine(event_end, dt.time(0, 0))
-                ) - dt.timedelta(minutes=1)
+        ics_urls = ics_url.split("|")
+        for ics_url in ics_urls:
+            response = requests.get(ics_url)
+            if response.ok:
+                cal = icalendar.Calendar.from_ical(response.text)
             else:
-                raise TypeError(f"Unknown type {type(event_start)} for DTSTART")
+                self.logger.error(f"Received an error when downloading ICS: {response.text}")
+                sys.exit(1)
 
-            new_event["isMultiday"] = (
-                new_event["endDatetime"] - new_event["startDatetime"]
-            ) > dt.timedelta(days=1)
+            cal_name = cal['X-WR-CALNAME']
```

Comment:
I'm not familiar with this property of ICS calendars. Is this field guaranteed to always exist or should we use  a fallback in case it doesn't? A quick research shows also other fields like `CALNAME`...?

Reply:
I added a better management, but I tested with all three supported calendar providers and they correctly display the name.